### ISSUE TITLE
Is it normal to found parameters in empty sizes?

### ISSUE THREAD

**Issue by tingxueronghua** [haotian-liu/LLaVA#302]
### Question

First, I am sure that the model could pretrain and can see the loss curve on wandb. However I want to find how to add extra token into the tokenizer and found something strange. Now I will show an example. Here I add the print function as follows to see the output of model parameters:

```
data_module = make_supervised_data_module(tokenizer=tokenizer, data_args=data_args)
print('previous:', model.model.layers[0].self_attn.q_proj.weight.data)
trainer = LLaVATrainer(model=model,
    tokenizer=tokenizer,
    args=training_args,
    **data_module)   
print('after:', model.model.layers[0].self_attn.q_proj.weight.data)

```

However, I found the output on all GPUs are empty. Is this normal?

```
previous: tensor([], device='cuda:0', dtype=torch.float16)
after: tensor([], device='cuda:0', dtype=torch.float16)
WARNING:root:Formatting inputs...Skip in lazy mode
previous: tensor([], device='cuda:1', dtype=torch.float16)
after: tensor([], device='cuda:1', dtype=torch.float16)
WARNING:root:Formatting inputs...Skip in lazy mode
previous: tensor([], device='cuda:2', dtype=torch.float16)
after: tensor([], device='cuda:2', dtype=torch.float16)
WARNING:root:Formatting inputs...Skip in lazy mode
previous: tensor([], device='cuda:4', dtype=torch.float16)
after: tensor([], device='cuda:4', dtype=torch.float16)
WARNING:root:Formatting inputs...Skip in lazy mode
previous: tensor([], device='cuda:7', dtype=torch.float16)
WARNING:root:Formatting inputs...Skip in lazy mode
previous: tensor([], device='cuda:3', dtype=torch.float16)
after: tensor([], device='cuda:7', dtype=torch.float16)
after: tensor([], device='cuda:3', dtype=torch.float16)
WARNING:root:Formatting inputs...Skip in lazy mode
previous: tensor([], device='cuda:5', dtype=torch.float16)
after: tensor([], device='cuda:5', dtype=torch.float16)
WARNING:root:Formatting inputs...Skip in lazy mode
previous: tensor([], device='cuda:6', dtype=torch.float16)
after: tensor([], device='cuda:6', dtype=torch.float16)
```

**Comment by tingxueronghua** [comment]
I am using deepspeed with zero3.json.

**Comment by haotian-liu** [comment]
Deepspeed zero3 uses meta device to automatically shard the model parameters.

To view the actual tensor sizes, you can use [zero2](https://github.com/haotian-liu/LLaVA/blob/main/scripts/zero2.json).

**Comment by tingxueronghua** [comment]
thanks for your quick reply! I will try your advice!
